probes/internal/utils/permissions/permissions.go

```diff
@@ -70,11 +71,11 @@ func CreateNegativeFinding(r checker.TokenPermission,
 
 	if r.File != nil {
 		f = f.WithLocation(r.File.Location())
-		f = f.WithRemediationMetadata(map[string]string{
-			"repo":     r.Remediation.Repo,
-			"branch":   r.Remediation.Branch,
-			"workflow": strings.TrimPrefix(f.Location.Path, ".github/workflows/"),
-		})
+	}
+	if metadata != nil {
+		updatedMetadata := metadata
```

Comment:
this mutates the original map.

Reply:
Do we need to copy the map at all? Could we do this instead?

```golang

workflowPath := strings.TrimPrefix(f.Location.Path, ".github/workflows/"
 f = f.WithRemediationMetadata(raw.Metadata.Metadata).WithWithRemediationMetadata("workflow", workflowPath 
```

Reply:
If that works, I'm fine with that instead

Reply:
Should be fixed by https://github.com/ossf/scorecard/pull/3816/commits/c16e38df6fd5552ee998a0f9fdf0ad5d72f746dd